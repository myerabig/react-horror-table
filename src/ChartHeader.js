import React from 'react';

function ChartHeader() {
  return (
    <tr>
      <th width="14%">Title</th>
      <th width="3%">Release Year</th>
      <th width="10%">Director</th>
      <th width="2%">IMDB Rating/10</th>
      <th width="10%">Subgenre</th>
      <th width="15%">Lead Actor(s)</th>
      <th width="44%">Plot Summary</th>
      <th width="2%">Runtime (minutes)</th>
    </tr>
  );
}

export default ChartHeader;
