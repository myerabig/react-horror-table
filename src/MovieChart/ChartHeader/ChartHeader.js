import React from 'react';
import SortArrows from '../SortArrows/SortArrows';

function ChartHeader(props) {
  return (
    <tr>
      <th width='16%'>
        Title
        <SortArrows
          movies={props.movies}
          setMovies={props.setMovies}
          property='title'
          property2='year'
          type='text'
        />
      </th>
      <th width='6%'>
        Release Year
        <SortArrows
          movies={props.movies}
          setMovies={props.setMovies}
          property='year'
          property2='title'
          type='number'
        />
      </th>
      <th width='12%'>
        Director
        <SortArrows
          movies={props.movies}
          setMovies={props.setMovies}
          property='director'
          property2='title'
          type='text'
        />
      </th>
      <th width='2%'>
        IMDB Rating/10
        <SortArrows
          movies={props.movies}
          setMovies={props.setMovies}
          property='imdbScore'
          property2='title'
          type='number'
        />
      </th>
      <th width='8%'>
        Subgenre
        <SortArrows
          movies={props.movies}
          setMovies={props.setMovies}
          property='subgenre'
          property2='title'
          type='text'
        />
      </th>
      <th width='20%'>Lead Actor(s)</th>
      <th width='45%'>Plot Summary</th>
      <th width='2%'>
        Runtime (minutes)
        <SortArrows
          movies={props.movies}
          setMovies={props.setMovies}
          property='runtime'
          property2='title'
          type='number'
        />
      </th>
    </tr>
  );
}

export default ChartHeader;
