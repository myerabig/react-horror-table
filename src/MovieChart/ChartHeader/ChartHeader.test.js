import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ChartHeader from './ChartHeader';

describe('ChartHeader', () => {
  afterEach(cleanup);

  let MOCK_MOVIES = [
    {
      title: 'a',
      year: '1'
    }
  ];
  const MOCK_SETMOVIES = movies => (MOCK_MOVIES = movies);

  it('should render as expected', () => {
    const { container } = render(
      <ChartHeader movies={MOCK_MOVIES} setMovies={MOCK_SETMOVIES} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
