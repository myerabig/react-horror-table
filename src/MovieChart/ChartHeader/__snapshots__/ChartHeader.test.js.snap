// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartHeader should render as expected 1`] = `
<tr>
  <th
    width="16%"
  >
    Title
    <div
      class="SortArrows"
    >
      <div
        align="right"
      >
        <img
          alt="Up"
          class="icon"
          id="upArrow"
          src="uparrow.png"
          width="20"
        />
        <img
          alt="Down"
          class="icon"
          id="downArrow"
          src="downarrow.png"
          width="20"
        />
      </div>
    </div>
  </th>
  <th
    width="6%"
  >
    Release Year
    <div
      class="SortArrows"
    >
      <div
        align="right"
      >
        <img
          alt="Up"
          class="icon"
          id="upArrow"
          src="uparrow.png"
          width="20"
        />
        <img
          alt="Down"
          class="icon"
          id="downArrow"
          src="downarrow.png"
          width="20"
        />
      </div>
    </div>
  </th>
  <th
    width="12%"
  >
    Director
    <div
      class="SortArrows"
    >
      <div
        align="right"
      >
        <img
          alt="Up"
          class="icon"
          id="upArrow"
          src="uparrow.png"
          width="20"
        />
        <img
          alt="Down"
          class="icon"
          id="downArrow"
          src="downarrow.png"
          width="20"
        />
      </div>
    </div>
  </th>
  <th
    width="2%"
  >
    IMDB Rating/10
    <div
      class="SortArrows"
    >
      <div
        align="right"
      >
        <img
          alt="Up"
          class="icon"
          id="upArrow"
          src="uparrow.png"
          width="20"
        />
        <img
          alt="Down"
          class="icon"
          id="downArrow"
          src="downarrow.png"
          width="20"
        />
      </div>
    </div>
  </th>
  <th
    width="8%"
  >
    Subgenre
    <div
      class="SortArrows"
    >
      <div
        align="right"
      >
        <img
          alt="Up"
          class="icon"
          id="upArrow"
          src="uparrow.png"
          width="20"
        />
        <img
          alt="Down"
          class="icon"
          id="downArrow"
          src="downarrow.png"
          width="20"
        />
      </div>
    </div>
  </th>
  <th
    width="15%"
  >
    Lead Actor(s)
  </th>
  <th
    width="50%"
  >
    Plot Summary
  </th>
  <th
    width="2%"
  >
    Runtime (minutes)
    <div
      class="SortArrows"
    >
      <div
        align="right"
      >
        <img
          alt="Up"
          class="icon"
          id="upArrow"
          src="uparrow.png"
          width="20"
        />
        <img
          alt="Down"
          class="icon"
          id="downArrow"
          src="downarrow.png"
          width="20"
        />
      </div>
    </div>
  </th>
</tr>
`;
